### PR TITLE
[SDRAN-1683,1714] Fixing memory leak issue and fixing repeated representation in oneof

### PR DESCRIPTION
**Pull request by eroshiva**
A big big gratitude to @kushal2705 for helping to fix the memory leak issue.

This PR also fixes the correect treatment of repeated structures inside oneof.

### CODE REVIEW

File: libasn1print/asn1print.c [hunk 1]
```diff
@@ -220,7 +220,7 @@ asn1print_module(asn1p_t *asn, asn1p_module_t *mod, enum asn1print_flags flags)
 			proto_enum_t **protoenum = calloc(0, sizeof(proto_enum_t *));
 			size_t protoenums = 0;
 			ret = asn1print_expr_proto(asn, mod, tc,
-									   message, &messages, protoenum, &protoenums,
+									   proto_module,
```
Comment (kushal2705) [inline libasn1print/asn1print.c]: below piece of code is not needed
proto_msg_t **message = calloc(0, sizeof(proto_msg_t *));
size_t messages = 0;
proto_enum_t **protoenum = calloc(0, sizeof(proto_enum_t *));
size_t protoenums = 0;

Comment (kushal2705) [reply libasn1print/asn1print.c]: Also, as discussed, below code can also be removed

if (messages > 0) {
				size_t existing_msgs = proto_module->messages;
				proto_module->message = realloc(proto_module->message,
												(existing_msgs + messages) * sizeof(proto_msg_t *));
				for (int i = 0; i < (int) messages; i++) {
					proto_module->message[existing_msgs + i] = message[i];
				}
				proto_module->messages = existing_msgs + messages;
			}
			if (protoenums > 0) {
				size_t existing_enums = proto_module->enums;
				proto_module->protoenum = realloc(proto_module->protoenum,
												  (existing_enums + protoenums) * sizeof(proto_enum_t *));
				for (int i = 0; i < (int) protoenums; i++) {
					proto_module->protoenum[existing_enums + i] = protoenum[i];
				}
				proto_module->enums = existing_enums + protoenums;
			}

File: libasn1print/asn1printproto.c [hunk 2]
```diff
@@ -1640,6 +1634,38 @@ proto_process_children(asn1p_expr_t *expr, proto_msg_t *msgdef, int repeated, in
 								se->expr_type, se->meta_type, expr->Identifier, se->Identifier);
 					}
 				}
+				if (oneof) {
```
Comment (kushal2705) [inline libasn1print/asn1printproto.c:1637]: why do we need this piece of code ?

**Review by kushal2705 — CHANGES_REQUESTED** [review]

File: libasn1print/asn1printproto.c (hunk 2 above)
Comment (eroshiva) [reply libasn1print/asn1printproto.c:1637]: This is to fix representation of SEQUENCE OF in Protobuf (it also requires nested messages).
This little part of code is not related to memory leak issue

File: libasn1print/asn1print.c (hunk 1 above)
Comment (eroshiva) [reply libasn1print/asn1print.c]: Indeed, forgot to remove

File: libasn1print/asn1printproto.c (hunk 2 above)
Comment (kushal2705) [reply libasn1print/asn1printproto.c:1637]: ok

**Review by kushal2705 — APPROVED** [review]

**Pull request merged**